veps/sig-storage/dic-on-heterogeneous-cluster/dic-on-heterogeneous-cluster.md

```diff
@@ -0,0 +1,656 @@
+- HCO: [kubevirt/hyperconverged-cluster-operator](https://github.com/kubevirt/hyperconverged-cluster-operator)
+- UI: [kubevirt-ui/kubevirt-plugin](https://github.com/kubevirt-ui/kubevirt-plugin)
+
+## Design
+To design a solution for heterogeneous clusters, the following key aspects must be addressed:
+
+- **Container disk image name**: For example, `"centos-stream9"`.
+- **Workload node architectures**: The architectures of the cluster nodes where the virtual machines (VMs) will run.
+- **Architectures supported by this image**: The architectures listed in the image manifest.
+
+### Responsibilities
+HCO manages the predefined golden images information, and is the configuration point for the custom golden images. That
+makes it responsible for the first aspect: the image names.
+
+HCO already watches the cluster nodes, and it makes sense to extend this functionality to monitor the current nodes,
+and maintain a list of supported architectures. That means that HCO will be responsible for the second aspect: the
+workload node architectures.
+
+The DataImportCron controller in CDI, is the component that monitors the images and that makes it to be responsible for
+the third aspect: the architectures supported by the image.
+
+### The DataImportCronTemplates file in the HCO image
+On HCO image build time, the predefined `DataImportCronTemplate` objects in the DataImportCronTemplates file will be
+annotated with the `ssp.kubevirt.io/dict.architectures` annotation, with a comma-separated list of the architectures
+listed in the specific image manifest.
+
+Currently, it will be a manual process. In a future phase we could automate by reading the image manifests at the image
+build time.
+
+### Maintaining a List of Supported Architectures
+The HCO will be enhanced to track the architectures of workload and control-plane nodes in the cluster. It will maintain
+two distinct lists of these architectures, derived from the `status.nodeInfo.architecture` field of each node, one for 
+the control-plane nodes and one for the workload nodes.
+
+These lists will be dynamically updated as nodes are added or removed from the cluster.
+
+By default, the HCO will identify workload nodes as those labeled with `node-role.kubernetes.io/worker`. If the
+`spec.workloads.nodePlacement` field in the `HyperConverged` CR is defined, HCO will use this field to determine the
+workload nodes instead.
+
+HCO will identify control-plane nodes as those labeled with `node-role.kubernetes.io/control-plane`.
+
+>**Note**: The control-plane architecture list is expected to always be with only one architecture, as the control-plane
+> nodes are expected to be homogeneous. HCO will not force this, though.
+
+### Reconciling the `SSP` Custom Resource
+The following changes in HCO behavior will be applied if:
+1. The new `enableMultiArchCommonBootImageImport` feature gate is enabled in the `HyperConverged` CR. This feature gate
```

Comment:
```suggestion
1. The new `EnableMultiArchCommonBootImageImport` feature gate is enabled in the `HyperConverged` CR. This feature gate
```

Don't these typically begin with a capital letter?

Reply:
HCO FGs are fields, rather than KubeVirt list of strings. yaml standard for filed names is to start with a low case .